cmds/core/chmod/chmod.go

```diff
@@ -42,6 +44,28 @@ func init() {
 		"use mode from reference file")
 }
 
+func changeMode(path string, operator string, octval uint64, mask uint64, mode os.FileMode) (err error) {
```

Comment:
I'd feel a bit more comfortable if you split out the 
calculate the mode given a string
from the
now go change fhe file
part. 
Possible?

Reply:
The existing mode of the file is an input to the change, in most cases. So changeMode needs a bit more logic than a plain os.Chmod. But maybe it can be simplified, I will try.

Reply:
I pushed an updated version that separates mode calculation from mode setting, as much as possible. Is this better?